### ISSUE TITLE
How to remove word from dictionary?

### ISSUE THREAD

**Issue by honghui-qiao** [Jason-Rev/vscode-spell-checker#117]
I added some wrong words to dictionary in vscode, how to remove them?

**Comment by Jason3S** [comment]
Good Question. I need to add this to the FAQ.

I wish this process was a bit easier.

It depends upon where you added it.

## User Settings
If you added it to your user settings, then open VS Code Settings: `F1 User Settings`

There should be a setting `cSpell.userWords` that looks something like this:

```javascript
"cSpell.userWords": [
    "CXLVIII",
    "DCCLXXXII",
    "Interglot",
    "MDCCLIV",
    "dicts",
    "tsmerge"
]
```

Remove the words you do not want in the list.

## Project Settings
Project settings are generally stored in `.vscode/cSpell.json` found in the root of you project directory.

Look for the `"words"` entry:
```javascript
    // words - list of words to be always considered correct
    "words": [
        "appname"
    ],
```

**Comment by susumuasaga** [comment]
I inadvertently added "routerlink" to global dictionary instead of camel case "routerLink". I wanted to remove this, but cannot find it in user settings:
```
  // User words to add to dictionary.  Should only be in the user settings.
  "cSpell.userWords": [],
```
and there is no .vscode/cSpell.json in the root of my project directory.

**Comment by Jason3S** [comment]
I have added a command to remove words from user settings: `F1` `Remove Words From ...`

![image](https://user-images.githubusercontent.com/3740137/38453511-3397299a-3a57-11e8-94af-4f46ecb544dc.png)

**Comment by talbct** [comment]
What about removing words from the built in dictionary?
Or even just explicitly marking a word as incorrect in order to maintain consistency?
For example, en-GB accepts `personalised` as well as `personalized`. While I am not sure whether this is correct or not, my project uses `personalised` and I would like to keep it consistent. Any way to do that?

**Comment by Jason3S** [comment]
This can be done by adding words to `flagWords`.

`cspell.json`
```json
"flagWords": [
    "personalized"
]
```

VS Code Settings:
```json
"cspell.flagWords": [
    "personalized"
]
```

**Comment by Leo7654** [comment]
"cSpell.flagWords": [
    "personalized"
]

**Comment by Jason3S** [comment]
Thank you, I adjusted my response.

**Comment by alanwilter** [comment]
> @talbct, This can be done by adding words to `flagWords`.
> 
> `cspell.json`
> 
> ```json
> "flagWords": [
>     "personalized"
> ]
> ```
> 
> VS Code Settings:
> 
> ```json
> "cSpell.flagWords": [
>     "personalized"
> ]
> ```

Fantastic plugin, yet, this is not a satisfactory solution, I'm not going to start flagging the whole US dictionary when I could simply have it removed and kept just the GB one. Could we have more sensible solution here please?

**Comment by Jason3S** [comment]
@alanwilter,

The issue is the most of the words are in the Oxford English dictionary. Setting the `"language": "en-GB"` will only include the Oxford dictionary. But, the Oxford English dictionary allows for a lot of American spellings. In a few cases, the `softwareTerms` dictionary has some American spellings.

Assuming you have NodeJS installed on your machine, the following command line will tell you which dictionaries contain a word:

```
npx cspell trace color
```
![image](https://user-images.githubusercontent.com/3740137/145726942-d7a65b24-4ba7-4add-b1c9-c8be1b220778.png)

Shows that `color` is in the US dictionary but not in the GB dictionary.

Another option to tell the spell check to forbid a word is to have a dictionary file and precede the words you do not want with `!`:

**`forbidden-american-words.txt`**

```txt
!color
!personaized
```
